### PR TITLE
feat(adapter-pg,adapter-neon): Enforce usage of connection pool (#22687)

### PR DESCRIPTION
**Pull request by SevInf**
PR was already reviewed in #22687, this iteration just
fixes PG import without breaking engine executor.

### CODE REVIEW

**Comment by SevInf** [comment]
confirmed that engines are fine here https://github.com/prisma/prisma-engines/pull/4655

File: helpers/test/presets/default.js
Comment (millsp) [inline helpers/test/presets/default.js:1]: Not sure why this is showing up again, wasn't this added already?

Comment (SevInf) [reply helpers/test/presets/default.js:1]: We reverted the PR it was added in

**Review by Jolg42 — APPROVED** [review]

File: packages/adapter-neon/src/neon.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /* eslint-disable @typescript-eslint/require-await */
-import type neon from '@neondatabase/serverless'
+import * as neon from '@neondatabase/serverless'
```
Comment (Jolg42) [inline packages/adapter-neon/src/neon.ts:2]: The 1 line change?

**Review by millsp — APPROVED** [review]

File: packages/adapter-pg/src/pg.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import type {
   TransactionOptions,
 } from '@prisma/driver-adapter-utils'
 import { Debug, err, ok } from '@prisma/driver-adapter-utils'
-import type pg from 'pg'
+import pg from 'pg'
```
Comment (Jolg42) [inline packages/adapter-pg/src/pg.ts:14]: I think that was the change actually

File: packages/adapter-neon/src/neon.ts (hunk 1 above)
Comment (Jolg42) [reply packages/adapter-neon/src/neon.ts:2]: (Nope)

File: packages/adapter-pg/src/pg.ts (hunk 2 above)
Comment (Jolg42) [reply packages/adapter-pg/src/pg.ts:14]: Previously `import * as pg from 'pg'`

File: packages/adapter-neon/src/neon.ts (hunk 1 above)
Comment (jkomyno) [reply packages/adapter-neon/src/neon.ts:2]: Given we moved from `import * as pg from 'pg'` to `import pg from 'pg'`, shouldn't we do the same with `neon` as well?

**Pull request merged**